container-search/src/main/java/com/yahoo/search/Query.java

```diff
@@ -299,14 +311,21 @@ public Query(HttpRequest request, CompiledQueryProfile queryProfile) {
     }
 
     /**
-     * Creates a query from a request
+     * Creates a query from a request containing a JSON-query.
      *
-     * @param request the HTTP request from which this is created
+     * @param request the HTTP request from which this is created.
+     * @param requestMap the property map of the query.
+     * @param queryProfile the query profile to use for this query, or null if none.
      */
-    public Query(HttpRequest request) {
-        this(request, null);
+    public Query(HttpRequest request, Map<String, String> requestMap, CompiledQueryProfile queryProfile) {
+
+        super(new QueryPropertyAliases(propertyAliases));
+        this.httpRequest = request;
+        init(requestMap, queryProfile);
```

Comment:
Preferable if you let the above constructor call `this(request, request.propertyMap(), queryProfile)`.